### PR TITLE
Docs: add info on export format to internals section

### LINKED ISSUE #4037: Docs: Internals: Data storage (Export archive)

### PR DESCRIPTION
**Pull request by ramirezfranciscof**
Fix  #4037
Partially addresses #4450

### CODE REVIEW

**Comment by giovannipizzi** [comment]
Hi Francisco, can you also remove the old content from the previous source (I think it was in `import_export/main.rst`), either only the content you moved, or all the file, so at the end we know what we didn't migrate yet?

**Comment by ramirezfranciscof** [comment]
Ok; my original idea was that I could anyways wait until all associated new content was merged and then delete all files in a given "remove" issue at once (seemed better than have each one decide what part to remove of the files), but from the message in the slack and this comment I assume you prefer that each deletes the part that each copies.

PD: It seems that we are having a BIG bottleneck in the tests. Maybe we should discuss if there is something that can be done about this in the afternoon meeting...

**Comment by ramirezfranciscof** [comment]
I updated all references to "export" into "AiiDA archive". There are some inconsitencies due to the references in the actual code (`export_version` inside the `metadata.json`, or the command being `verdi export migrate`) but I don't think there is much I can do about those.

Now should be ready to review @giovannipizzi @sphuber

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:26]:
```suggestion
An AiiDA archive file is a file usually ending with extension ``.aiida``, typically compressed in ``.zip`` or ``.tar.gz`` format, with the following content:
```

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:32]: You might want to use a longer reference name using the full prefix of where this is - these are global labels. So e.g. something like `_internal_architecture:orm:archive:metadata-json:`

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:60]:
```suggestion
The entities that are exported are mentioned in `unique_identifiers` with their respective unique identifiers.
```

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:48]:
```suggestion
At the beginning of the file, we see the version of the archive file (under ``export_version``) and the version of the AiiDA code.
```

Comment (giovannipizzi) [reply docs/source/internals/data_storage.rst:48]: I think here and below sometimes single backpacks are used, sometimes stars etc - I suggest using double backticks everywhere we refer to a literal string (e.g. the key of one of the JSON files) - I didn't put a comment on each of them

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [reply docs/source/internals/data_storage.rst:60]: Same for the fields below

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:77]: As above

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:88]:
```suggestion
For every link the UUIDs of the connected nodes, as well as the name of the link, are mentioned.
```
(I think people should know by now what the UUID of a node is)

File: docs/source/internals/data_storage.rst
Comment (giovannipizzi) [inline docs/source/internals/data_storage.rst:95]:
```suggestion
This ensures that the archive files are compatible between different actual database backend/schema implementations (e.g. Django, SQLAlchemy, ...).
```

**Review by giovannipizzi — CHANGES_REQUESTED** [review]
Some minor suggestions.

Also, please add a redirect from the old URL describing the data storage to this new URL.

**Comment by ramirezfranciscof** [comment]
@giovannipizzi for now the redirection works for the full page (so `import_export/main.html --> internals/data_storage.html`), I have just asked if it was possible to do section redirection (`import_export/main.html#export-file-format --> internals/data_storage.html#aiida-archive-format`) which is the link that you originally wanted to fix.

**Comment by ramirezfranciscof** [comment]
For which the answer is that apparently this can't be done, so I guess this is the best we can do.

**Review by giovannipizzi — APPROVED** [review]
Thanks!

**Pull request merged**
